src/js/client/editprefset-prefSettingAdjuster-widget.js

```diff
@@ -0,0 +1,269 @@
+        "current.blank": {
+            func: "{that}.updateBlankDisabling"
+        },
+        "current.value": [{
+            funcName: "gpii.devpmt.prefSettingAdjuster.updateBooleanValueLabel",
+            args: ["{that}"]
+        }]
+    }
+});
+
+gpii.devpmt.prefSettingAdjuster.updateBooleanValueLabel = function (that) {
+    // if (that.model.metadata.schema.type === "boolean") {
+    jQuery("#pmt-boolean-value").css("display","inline");
+    jQuery("#pmt-boolean-value").css("color","black");
+    if (that.model.current.value === true)  {
+        jQuery("#pmt-boolean-value").html("true");
+    }
+    else if (that.model.current.value === false) {
+        jQuery("#pmt-boolean-value").html("false");
+    }
+};
+
+/**
+ * Updates the UI based on whether Blank is selected.
+ * If blank is selected, then the rest of the editing
+ * apparatus is disabled.
+ */
+gpii.devpmt.prefSettingAdjuster.updateBlankDisabling = function (that) {
+    var disable = that.model.current.blank;
+    that.dom.locate("valueInput").prop("disabled", disable);
+};
+
+/**
+ * watchInputKeys - Watches value input and saves
+ * on Enter.
+ *
+ * @param (Object) that - prefSettingAdjuster
+ * @param (Object) e - jQuery Event
+ */
+gpii.devpmt.prefSettingAdjuster.watchInputKeys = function (that, e) {
+    // Save the value when the enter key is pressed
+    if (e.keyCode === 13) {
+        that.saveUpdateValue();
+    };
+};
+
+/**
+ * Save/Update using the value the user has input. This is
+ * typically caused by clicking the Ok button.
+ */
+gpii.devpmt.prefSettingAdjuster.saveUpdateValue = function (that, devpmt) {
+    that.applier.change("active", false);
+    var segs = ["contexts", that.model.current.context, "preferences"];
+    if (that.model.current.product) {
+        segs.push(that.model.current.product.replace(/\./g, "\\."), that.model.current.term.replace(/\./g, "\\."));
+    }
+    else {
+        segs.push(that.model.current.term.replace(/\./g, "\\."));
+    }
+    var path = ["flatPrefs"].concat(segs).join(".");
+    // If the `blank` checkbox is ticked than we are actually going
+    // to delete this key, rather than update it.
+    if (that.model.current.blank) {
+        devpmt.applier.change(path, false, "DELETE");
+        devpmt.addEditToUnsavedList("Removed setting " + that.model.metadata.name +
+                " in context " + that.model.current.context);
+    }
+    else {
+        var newValue = that.model.current.value; //that.dom.locate("valueInput").val();
+        devpmt.applier.change(path, newValue);
+        devpmt.addEditToUnsavedList("Changed setting " + that.model.metadata.name +
+                " in context " + that.model.current.context + " to " + newValue);
+    }
+    // In the event that we didn't actually change any values, the model listener
+    // won't rerender the entire page.
+    that.renderInitialMarkup();
+    // Remove the active editing highlight
+    $(".pmt-value-display").removeClass("pmt-value-editing");
```

Comment:
Same as above, please use "selectors" block to hold selectors. Same comment applies to line 268 and other places. Thanks.

Reply:
updated